test/integration/allocatable_test.go

```diff
@@ -0,0 +1,194 @@
+		v1.ResourceCPU:    *resource.NewMilliQuantity(500, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI),
+	}
+	node.Status.Capacity = v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewMilliQuantity(500, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI),
+	}
+	node, err = cs.CoreV1().Nodes().Create(context.TODO(), node, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create Node %q: %v", nodeName2, err)
+	}
+
+	// Create a bigger Node.
+	nodeName3 := "fake-node-big"
+	node = st.MakeNode().Name(nodeName3).Label("node", nodeName3).Obj()
+	node.Status.Allocatable = v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewMilliQuantity(500, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(5100, resource.DecimalSI),
+	}
+	node.Status.Capacity = v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewMilliQuantity(500, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(5100, resource.DecimalSI),
+	}
+	node, err = cs.CoreV1().Nodes().Create(context.TODO(), node, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create Node %q: %v", nodeName3, err)
+	}
+
+	// Create Pods.
+	var pods []*v1.Pod
+	podNames := []string{"small-1", "small-2", "small-3", "small-4"}
+	pause := imageutils.GetPauseImageName()
+	for i := 0; i < len(podNames); i++ {
+		pod := st.MakePod().Namespace(ns).Name(podNames[i]).Container(pause).Obj()
+		pod.Spec.Containers[0].Resources = v1.ResourceRequirements{
+			Requests: v1.ResourceList{
+				v1.ResourceMemory: *resource.NewQuantity(100, resource.DecimalSI),
+			},
+		}
+		pods = append(pods, pod)
+	}
+
+	// Make a big pod.
+	podNames = append(podNames, "big-1")
+	pod := st.MakePod().Namespace(ns).Name("big-1").Container(pause).Obj()
+	pod.Spec.Containers[0].Resources = v1.ResourceRequirements{
+		Requests: v1.ResourceList{
+			v1.ResourceMemory: *resource.NewQuantity(5000, resource.DecimalSI),
+		},
+	}
+	pods = append(pods, pod)
+
+	// Create the Pods. By default the small pods should land on the small nodes.
+	t.Logf("Start to create 5 Pods.")
+	for i := range pods {
+		t.Logf("Creating Pod %q", pods[i].Name)
+		_, err = cs.CoreV1().Pods(ns).Create(context.TODO(), pods[i], metav1.CreateOptions{})
+		if err != nil {
+			t.Fatalf("Failed to create Pod %q: %v", pods[i].Name, err)
+		}
+	}
+
+	// Wait for all Pods scheduled.
+	err = wait.Poll(time.Millisecond*200, wait.ForeverTestTimeout, func() (bool, error) {
+		if len(testCtx.Scheduler.SchedulingQueue.PendingPods()) == 0 {
+			return true, nil
+		}
+		t.Logf("%d pending pods: %v", len(testCtx.Scheduler.SchedulingQueue.PendingPods()), testCtx.Scheduler.SchedulingQueue.PendingPods())
+		return false, nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
It can be flaky to rely on `len(pendingPods)` to tell if all pods get scheduled. I'd suggest to remove this block. Instead, use wait.Poll() below to ensure each pod gets scheduled within a timeout.

Reply:
done